nbbuild/antsrc/org/netbeans/nbbuild/extlibs/MavenSkeletonProject.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.nbbuild.extlibs;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.apache.tools.ant.BuildException;
+import org.apache.tools.ant.Task;
+
+/**
+ * Task to check that external libraries have legitimate licenses, etc.
+ */
+public class MavenSkeletonProject extends Task {
+
+    private File nball;
+
+    public void setNball(File nball) {
+        this.nball = nball;
+    }
+
+    private File mavenfolderFile;
+
+    /**
+     * JUnit-format XML result file to generate, rather than halting the build.
+     */
+    public void setMavenFolder(File report) {
+        this.mavenfolderFile = report;
+    }
+
+    private Set<String> modules;
+
+    public @Override
+    void execute() throws BuildException {
+        try { // XXX workaround for http://issues.apache.org/bugzilla/show_bug.cgi?id=43398
+            if (getProject().getProperty("allmodules") != null) {
+                modules = new TreeSet<>(Arrays.asList(getProject().getProperty("allmodules").split("[, ]+")));
+                modules.add("nbbuild");
+            } else {
+                Path nbAllPath = nball.toPath();
+                modules = new TreeSet<>(
+                        Files.walk(nbAllPath)
```

Comment:
most of the streams which are returned from `Files.*` must be closed after use. Would be good to wrap the stream in an arm-block and do the rest in the block.

I have some jackpot rules here (since its easy to forget - most streams don't have to be closed):
https://github.com/mbien/jackpot-inspections/blob/5b518358408dda4008be5b7f2602aa3304a0c8fc/ProbableBugs.hint#L32-L40